transport-udt/src/main/java/io/netty/channel/udt/nio/NioUdtByteConnectorChannel.java

```diff
@@ -150,6 +151,11 @@ protected int doWriteBytes(final ByteBuf byteBuf) throws Exception {
     }
 
     @Override
+    protected ChannelFuture shutdownInput() {
+        return voidPromise();
```

Comment:
I think this is not correct as voidPromise() will fail when you try to add listeners to it. So better just return a `newFailedFuture()` ?